# Abhinav Adduri
<img src="assets/images/abhinav-portrait.jpg" align="right" alt="Abhinav Adduri"/>

[\[LinkedIn\]](https://www.linkedin.com/in/abhinadduri/) &nbsp; [\[aadduri [AT] cs.cmu.edu\]](mailto:aadduri@cs.cmu.edu) 

Hello! I am a 5th year Ph.D. Candidate in Computer Science at CMU advised by [Dr. Hosein Mohimani](http://mohimanilab.cbd.cmu.edu/) and a Felicis Fellow in AI. My research leverages protein language models to combat the rising rates of antimicrobial resistance by accelerating the discovery of antimicrobials with novel mechanisms of action (MOA).

Downstream of drug discovery, my work also focuses on representation learning and aligning modalities (e.g., small molecule drug and protein target) to predict drug-target interactions and mechanism of action.

Organisms in nature such as bacteria or fungi can produce drugs that can benefit humanity, but as of now, finding these drugs is laborious and mostly trial and error. Our work combines protein language models with mass spectrometry analysis to predict the drug-like molecules these microbes synthesize, and to characterize their bioactivity.

We are working to publish our most recent discovery: an antifungal that we've validated in mouse models that kills *Candida Auris*, a drug-resistant fungal strain. We are especially excited about this compound as the CDC recently deemed *C. Auris* an urgent threat. 

Before starting my Ph.D., I majored in Computer Science at UC Berkeley. In my free time, I like to go outdoor rock climbing.

# Publications in Review
McNutt, Andrew.\*, **Adduri, A.\***, Ellington, C.\*, ..., Xing, E., Mohimani, H. Koes, David. SPRINT Enables Interpretable and Fast Virtual Screening against Thousands of Proteomes. Under review at MLSB NeurIPS workshop.

Behsaz, B.\*, **Adduri, A.\***, Guler, M.\*, ..., & Mohimani, H. Pathogen-oriented platform for large-scale natural product discovery identifies novel antifungal targeting drug-resistant candidiasis. *Under review at Nature Communications.*

**Adduri, A.**, ..., Xing, E., Koes, D., & Mohimani, H. Interpretable adenylation domain specificity prediction using protein language models. *Under review at PNAS.*

Ellington, C., Lengerich, B., Watkins, T., Yang, J., **Adduri, A.**, Xiao, H., Kellis, M., Xing, E. Contextualized networks reveal heterogeneous transcriptomic regulation in tumors at sample-specific resolution. *Under review at PNAS.*

*Equal Contribution.

# Publications
Ellington, C.\*, **Adduri, A.\***, ..., Mohimani, H. Xing, E. Ultra-High-Throughput Virtual Screening for Antimicrobials. Machine Learning in Computational Biology 2024.

**Adduri, A.**, & Kim, S. (2024). Ornaments for accurate and efficient allele-specific expression estimation with bias correction. *Manuscript accepted in The American Journal of Human Genetics.* 

Yan, D.\*, Zhou, M.\*, **Adduri, A.\***, ..., & Mohimani, H. (2024). Seq2PKS: Discovering modular type I cis-AT polyketide natural products by integrating computational mass spectrometry and genome mining. *Nature Communications, 15.1 (2024): 5356.*

Mongia, M., Baral, R., **Adduri, A.**, Yan, D., Liu, Y., Bian, Y., Kim, P., Behsaz, B., & Mohimani, H. (2023). AdenPredictor: Accurate prediction of the adenylation domain specificity of nonribosomal peptide biosynthetic gene clusters in microbial genomes. *Bioinformatics, 39(Supplement 1), i40-i46.*

**Adduri, A.** (2019). IoT integration in manufacturing processes. In *Simulation for Industry 4.0: Past, Present, and Future* (pp. 129-139).

**Adduri, A.**, & Schruben, L. (2017). The Tao of Simulation. In *2017 Winter Simulation Conference (WSC)* (pp. 607-616). IEEE.

*Equal Contribution.

# Prior Work

|     Company      |       Location       |             Dates            |         My Contributions         |
|------------------|----------------------|------------------------------|-------------------------|
| Mozilla          | Portland, OR         |    May 2017 - August 2017    | Designed and wrote [Firefox Send](https://github.com/mozilla/send), a peer-to-peer, private file sharing app with 13.2k stars and 1.6k forks on github. Send was covered by [Forbes](https://www.forbes.com/sites/leemathews/2017/08/04/firefox-makers-new-app-makes-secure-file-sharing-a-breeze/?sh=65fd86b71d03), TechRepublic, etc, but was eventually shutdown in 2020 due to [malicious users](https://en.wikipedia.org/wiki/Firefox_Send). The engine I wrote has been reused in multiple open source projects, such as [ffsend](https://github.com/timvisee/ffsend), and Mozilla has since resurrected the project as [Thunderbird Send](https://addons.thunderbird.net/en-us/thunderbird/addon/filelink-provider-for-send/).       |
| Rubrik           | Palo Alto, CA        |    May 2018 - August 2018    | Designed and implemented a distributed cache system for the filesystems team, and designed RPCs to expose the node cache space.                                     |
| Square / Caviar  | San Francisco, CA    |    May 2019 - August 2019    | Improved pre-checkout food delivery ETA predictions, and improved recall / precision of fraud detection models by 3.5x / 2.5x respectively.                         |

# Awards

Felicis Fellows in AI (2024) - a fellowship program that connects students in AI with industry leaders and startup founders.

CMU SCS Graduate Student Service Award Winner

# Patents

[Automated Review of Source Code for Style Issues](https://patents.justia.com/patent/11294792), Rubrik 2021

# Other Projects
[Ultra-Fast DTI](https://github.com/abhinadduri/) - a co-embedding model for pan-species drug-protein interaction predictions, screening a drug against 4.3 million protein targets across evolutionary space in milliseconds.

[MASPR](https://github.com/abhinadduri/MASPR) - a multi-task learning framework that leverages protein language models for predicting adenylation domain specificity in nonribosomal peptide synthetases.

[NPDiscover](https://run.npanalysis.org/) - a web server for our work on predicting drug-like secondary metabolites from microbial genomes.

[Ornaments](https://github.com/SeyoungKimLab/Ornaments) - a lightweight modification of kallisto that for variant-aware pseudoalignment of RNA-Seq reads to obtain expected allele-specific read counts at heterozygous variant loci.

[RecursiveGPT](https://github.com/james-julius/recursive-gpt) - a chat interface I helped develop at an [AGI house](https://agihouse.ai/) hackathon which searches for adjacent GPT conversations in embedding space and suggests follow-up questions to enhance your knowledge dives.

[kilter.net](https://github.com/mdayao/kilter.net) - a convolutional variational autoencoder to generate climbing routes conditioned on difficulty and board angle.

[CMU Coffee Chats](https://github.com/scs-phd-deans-committee/coffee-chats-website) - a simple website to connect like-minded CMU students for conversations over coffee â˜•.

[Tao](https://github.com/abhinadduri/tao) - a browser based discrete event simulation engine with IoT capabilities.
