# Abhinav Adduri
[\[LinkedIn\]](https://www.linkedin.com/in/abhinadduri/) &nbsp; [\[CV\]](https://raw.githubusercontent.com/abhinadduri/abhinadduri.github.io/main/assets/Abhinav_CV.pdf) &nbsp; [\[aadduri@andrew.cmu.edu\]](mailto:aadduri@andrew.cmu.edu) 

<img src="assets/images/abhinav-portrait.jpg" align="right" alt="Abhinav Adduri"/>

Hello! I am a 5th year Ph.D. Candidate in Computer Science at CMU advised by [Dr. Hosein Mohimani](http://mohimanilab.cbd.cmu.edu/). My research is in applied deep learning and building and using foundation models to accelerate natural product discovery and drug discovery.

Organisms in nature such as bacteria or fungi can produce drugs that can benefit humanity, but as of now, finding these drugs is laborious and mostly trial and error. My work uses LLMs and bioinformatics algorithms to reliably predict the drug-like compounds these microbes can create, and characterize whether they will be useful to humans as antimicrobials or as cancer therapeutics.

We are working to publish our most recent discovery: an antifungal that we've validated in mouse models that kills a drug-resistant 
fungal strain that the CDC considers an urgent threat. To address the need for high-throughput performance, we make heavy use of [Rust](https://www.rust-lang.org), which in tandem with efficient algorithm design has led to order-of-magnitude improvements in runtimes.

I am also interested in using self-supervised techniques on multi-modal biological data to tackle problems downstream of drug discovery, such as drug-target interaction prediction and mass spectrum-molecule matching.

Before starting my Ph.D., I majored in Computer Science at UC Berkeley. In my free time, I like to go outdoor rock climbing or play board games.

# Publications
Mongia, M., Baral, R., **Adduri, A.**, Yan, D., Liu, Y., Bian, Y., Kim, P., Behsaz, B., & Mohimani, H. (2023). AdenPredictor: Accurate prediction of the adenylation domain specificity of nonribosomal peptide biosynthetic gene clusters in microbial genomes. *Bioinformatics, 39*(Supplement 1), i40-i46.

**Adduri, A.** (2019). IoT integration in manufacturing processes. In *Simulation for Industry 4.0: Past, Present, and Future* (pp. 129-139).

**Adduri, A.**, & Schruben, L. (2017). The Tao of Simulation. In *2017 Winter Simulation Conference (WSC)* (pp. 607-616). IEEE.

*Equal Contribution.

# Pending Publications
Behsaz, B.\*, **Adduri, A.\***, Guler, M.\*, ..., & Mohimani, H. (Expected 2024). Pathogen-oriented platform for large-scale natural product discovery identifies novel antifungal targeting drug-resistant candidiasis. *Manuscript under review at Nature Biotechnology.*

**Adduri, A.**, & Kim, S. (2023). Ornaments for accurate and efficient allele-specific expression estimation with bias correction. *Manuscript awaiting final decision in The American Journal of Human Genetics. [[ArXiv](https://www.biorxiv.org/content/10.1101/2023.10.25.564046v1.full)]*

Yan, D.\*, Zhou, M.\*, **Adduri, A.\***, Guler, M., Liu, S., Shin, H., Kovach, T., Oh, G., Liu, X., Deng, Y., Cao, L., Behsaz, B., & Mohimani, H. (Expected 2024). Seq2PKS: Discovering modular type I cis-AT polyketide natural products by integrating computational mass spectrometry and genome mining. *Manuscript awaiting final decision in Nature Communications.*

**Adduri, A.**, McNutt, A., Yan, D., Guler, M., Suraparaju, C., Fang, N., Bodden, C., Koes, D., Tripathi, A., & Mohimani, H. (Expected 2024). Multi-task learning with protein language models enables interpretable adenylation domain specificity prediction. *Manuscript in preparation for submission.*

*Equal Contribution.

# Prior Work

|     Company      |       Location       |             Dates            |         My Contributions         |
|------------------|----------------------|------------------------------|-------------------------|
| Mozilla          | Portland, OR         |    May 2017 - August 2017    | Designed and wrote [Firefox Send](https://github.com/mozilla/send), a peer-to-peer, private file sharing app with 13.2k stars and 1.6k forks on github. Send was covered by [Forbes](https://www.forbes.com/sites/leemathews/2017/08/04/firefox-makers-new-app-makes-secure-file-sharing-a-breeze/?sh=65fd86b71d03), TechRepublic, etc, but was eventually shutdown in 2020 due to [malicious users](https://en.wikipedia.org/wiki/Firefox_Send). The engine I wrote has been reused in multiple open source projects, such as [ffsend](https://github.com/timvisee/ffsend), and Mozilla has since resurrected the project as [Thunderbird Send](https://addons.thunderbird.net/en-us/thunderbird/addon/filelink-provider-for-send/).       |
| Rubrik           | Palo Alto, CA        |    May 2018 - August 2018    | Designed and implemented a distributed cache system for the filesystems team, and designed RPCs to expose the node cache space.                                     |
| Square / Caviar  | San Francisco, CA    |    May 2019 - August 2019    | Improved pre-checkout food delivery ETA predictions, and improved recall / precision of fraud detection models by 3.5x / 2.5x respectively.                         |

# Patents

[Automated Review of Source Code for Style Issues](https://patents.justia.com/patent/11294792), Rubrik 2021

# Other Projects
[RecursiveGPT](https://github.com/james-julius/recursive-gpt) - a chat interface I helped develop at an [AGI house](https://agihouse.ai/) hackathon which searches for adjacent GPT conversations in embedding space and suggests follow-up questions to enhance your knowledge dives.

[kilter.net](https://github.com/mdayao/kilter.net) - a convolutional variational autoencoder to generate climbing routes conditioned on difficulty and board angle.

[CMU Coffee Chats](https://github.com/scs-phd-deans-committee/coffee-chats-website) - a simple website to connect like-minded CMU students for conversations over coffee â˜•.

[Tao](https://github.com/abhinadduri/tao) - a browser based discrete event simulation engine with IoT capabilities.

